### PR TITLE
add proper validation tests

### PR DESCRIPTION
**Pull request by notwaldorf**
I've added most of the test cases from [here](http://blogs.msdn.com/b/testing123/archive/2009/02/05/email-address-test-cases.aspx). I've ignored:
- the ones that check for an invalid ip, since that's not a regex' job
- the ones that allow "" around the address and [ ] around the IP, since RFC 5322 has given up on those as well.

Initially some tests failed, so I've added the ^ and the $ to specify the start/end of input. Now the all pass. :tada: 

/cc @cdata @morethanreal @azakus

### CODE REVIEW

**Comment by notwaldorf** [comment]
Alright. I have taken @garlicnation's suggestion of adding a helper method to minimize the copy pasta, but I have kept the tests as separate ones, since they're better to see what actually went wrong.

@cdata @garlicnation PTAL

File: email-validator.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
     ],
 
     validate: function(value) {
-      var re = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/i;
+      var re = /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/i;
```
Comment (cdata) [inline email-validator.html]: image

**Comment by cdata** [comment]
Me likey :+1:

**Comment by garlicnation** [comment]
LGTM.

**Comment by notwaldorf** [comment]
:tada: Thanks!

**Pull request merged**